snapcraft/application.py

```diff
@@ -213,6 +234,7 @@ def _setup_partitions(self, yaml_data: dict[str, Any]) -> list[str] | None:
         # This is why we have the enablement after the check, if not
         # we could have directly returned with .get_partitions() which
         # handles the empty case.
+        craft_parts.Features.reset()
         craft_parts.Features(enable_partitions=True)
```

Comment:
This is another upstream nuisance.  Before this runs, craft-application checks for features and unintentionally defines an empty feature set:

https://github.com/canonical/craft-application/blob/1509cec9af705c762b06725bf867ef1d9e1d822b/craft_application/commands/lifecycle.py#L44-L45